src/augmentation.py

```diff
@@ -166,4 +166,166 @@ def _perspective_transform_augment_images(self, images, random_state, parents, h
+            return False
+
+
+class RandomCropFixedSize(iaa.Augmenter):
+    def __init__(self, px=None, name=None, deterministic=False, random_state=None):
+        super(RandomCropFixedSize, self).__init__(name=name, deterministic=deterministic, random_state=random_state)
+        self.px = px
+        if isinstance(self.px, tuple):
+            self.px_h, self.px_w = self.px
+        elif isinstance(self.px, int):
+            self.px_h = self.px
+            self.px_w = self.px
+        else:
+            raise NotImplementedError
+
+    def _augment_images(self, images, random_state, parents, hooks):
+
+        result = []
+        seeds = random_state.randint(0, 10 ** 6, (len(images),))
+        for i, image in enumerate(images):
+            seed = seeds[i]
+            image_cr = self._random_crop(seed, image)
+            result.append(image_cr)
+        return result
+
+    def _augment_keypoints(self, keypoints_on_images, random_state, parents, hooks):
+        result = []
+        return result
+
+    def _random_crop(self, seed, image):
+        height, width = image.shape[:2]
+
+        np.random.seed(seed)
+        if height > self.px_h:
+            crop_top = np.random.randint(height - self.px_h)
+        elif height == self.px_h:
+            crop_top = 0
+        else:
+            raise ValueError("To big crop height")
+        crop_bottom = crop_top + self.px_h
+
+        np.random.seed(seed + 1)
+        if width > self.px_w:
+            crop_left = np.random.randint(width - self.px_w)
+        elif width == self.px_w:
+            crop_left = 0
+        else:
+            raise ValueError("To big crop width")
+        crop_right = crop_left + self.px_w
+
+        if len(image.shape) == 2:
+            image_cropped = image[crop_top:crop_bottom, crop_left:crop_right]
+        else:
+            image_cropped = image[crop_top:crop_bottom, crop_left:crop_right, :]
+        return image_cropped
+
+    def get_parameters(self):
+        return []
+
+
+class PadToFitInNetwork(iaa.Augmenter):
+    def __init__(self, divisor=2, pad_mode='symmetric', name=None, deterministic=False, random_state=None):
+        super(PadToFitInNetwork, self).__init__(name=name, deterministic=deterministic, random_state=random_state)
+        self.divisor = divisor
+        self.pad_mode = pad_mode
+
+    def _augment_keypoints(self, keypoints_on_images, random_state, parents, hooks):
+        return keypoints_on_images
+
+    def _augment_images(self, images, random_state, parents, hooks):
+
+        result = []
+        for i, image in enumerate(images):
+            image_padded = self._pad_image(image)
+            result.append(image_padded)
+        return result
+
+    def _pad_image(self, image):
+        height = image.shape[0]
+        width = image.shape[1]
+
+        pad_vertical = self._get_pad(height)
+        pad_horizontal = self._get_pad(width)
+        pad_sequence = self._get_pad_sequence(pad_vertical, pad_horizontal)
```

Comment:
@apyskir I would hide `self._get_pad(height)` and `self._get_pad(width)` in the `self._get_pad_sequence` since we are not using it outside of that func

Reply:
@jakubczakon OK